sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/src/LruCache.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Data.SchemaRegistry.ApacheAvro
+{
+    internal class LruCache<K,V>
+    {
+        private readonly int _capacity;
+        private readonly LinkedList<KeyValuePair<K, V>> _linkedList;
+        private readonly Dictionary<K, LinkedListNode<KeyValuePair<K, V>>> _map;
+        private readonly object _syncLock;
+
+        public LruCache(int capacity)
+        {
+            _capacity = capacity;
+            _linkedList = new LinkedList<KeyValuePair<K, V>>();
+            _map = new Dictionary<K, LinkedListNode<KeyValuePair<K, V>>>();
+            _syncLock = new object();
+        }
+
+        public bool TryGet(K key, out V value)
+        {
+            lock (_syncLock)
+            {
+                if (_map.TryGetValue(key, out var node))
+                {
+                    value = node.Value.Value;
+                    _linkedList.Remove(node);
```

Comment:
Have you considered using the head as the most recent item?  It's normally quicker with a linked list to insert at the head than the end since you can skip walking the existing items.    I suspect that you'll read more frequently than you'll overflow capacity, though I could be ignorant of the scenario.

Reply:
There shouldn't be a perf difference as I believe there are pointers to the head and the tail and this is a doubly linked list.

Reply:
That said, I think I will swap it around for sake of clarity.

Reply:
Huh.... I had no idea that it was a doubly linked list.   I learned me something today.

Reply:
yep, otherwise that AddLast API would be a pretty mean thing to do to users.